src/Umbraco.Web.UI.Client/src/views/common/overlays/treepicker/treepicker.controller.js

```diff
@@ -20,16 +33,60 @@ angular.module("umbraco").controller("Umbraco.Overlays.TreePickerController",
             selectedSearchResults: []
         }
 
-        $scope.model.selection = [];
-
-        //Used for toggling an empty-state message
+        $scope.model.selection = [];
+
+        //Used for toggling an empty-state message
         //Some trees can have no items (dictionary & forms email templates)
         $scope.hasItems = true;
-        $scope.emptyStateMessage = dialogOptions.emptyStateMessage;
-
+        $scope.emptyStateMessage = dialogOptions.emptyStateMessage;
+
+        $scope.languages = [];
+        $scope.selectedLanguage = {};
+        $scope.page = {};
+        $scope.page.languageSelectorIsOpen = false;
+
+        var evts = [];
+
+        // Listen for language updates
+        evts.push(eventsService.on("editors.languages.languageDeleted",
+            function (e, args) {
+                languageResource.getAll().then(function (languages) {
+                    $scope.languages = languages;
+                });
+            }));
+
+        evts.push(eventsService.on("editors.languages.languageCreated",
+            function (e, args) {
+                languageResource.getAll().then(function (languages) {
+                    $scope.languages = languages;
+                });
+            }));
+
+        // load languages
+        languageResource.getAll().then(function (languages) {

```

Comment:
This should be done inside of the scope.$init method since that is where initialization occurs for this control